core/src/data_sources/data_source.rs

```diff
@@ -594,37 +596,149 @@ impl DataSource {
             utils::now() - now
         ));
 
-        // Chunk splits into a vectors of 8 chunks (Vec<Vec<String>>)
-        let chunked_splits = splits
-            .chunks(8)
-            .map(|chunk| chunk.to_vec())
-            .collect::<Vec<_>>();
+        let current_doc = store
+            .load_data_source_document(
+                &self.project,
+                &self.data_source_id(),
+                &document_id.to_string(),
+                &None,
+            )
+            .await?;
 
+        let mut should_embbed = match current_doc {
+            Some(current_doc) => current_doc.hash != document_hash,
+            None => true,
+        };
         // Ordered results with offset, stirng and vector
         let mut e: Vec<(usize, String, EmbedderVector)> = vec![];
-
-        let now = utils::now();
         let mut offset: usize = 0;
-        // Embed batched chunks sequentially.
-        for chunk in chunked_splits {
-            let r = EmbedderRequest::new(
-                self.config.provider_id.clone(),
-                &self.config.model_id,
-                chunk.iter().map(|s| s.as_str()).collect::<Vec<_>>(),
-                self.config.extras.clone(),
-            );
+        if should_embbed == false {
+            let scroll_results = qdrant_client
+                .scroll(&ScrollPoints {
+                    collection_name: self.qdrant_collection(),
+                    with_vectors: Some(true.into()),
+                    limit: Some(10000),
```

Comment:
I picked 10 000 here to give us some margin on the max chunk size we accept on the connector size (1MB).
1 000 000/256 = ~3900.

Let me know what you think.

Reply:
I don't think we need to put a limit since this is limited to one document. What happens if we exhaust that limit? We do embed the rest? that seems even more wasteful?

Reply:
Without a limit we only get a sub part of it, leading to always re-embedding.

Reply:
You mean there's a default limit?

Reply:
(default limit is 10 btw)

Reply:
why paginate if you need everything anyway, since we are not managing any ordering?

Reply:
>You mean there's a default limit?
Yes, default limit is 10.

Reply:
Agree :)

Reply:
I added support for pagination.